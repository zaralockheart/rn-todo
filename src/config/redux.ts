// Imports: Dependencies
import {applyMiddleware, combineReducers, createStore} from 'redux'
import thunk from 'redux-thunk'
import AsyncStorage from '@react-native-community/async-storage'
import {createLogger} from 'redux-logger'
import {persistReducer, persistStore} from 'redux-persist'
import {reducer as formReducer} from 'redux-form'

// Imports: Reducers

// Redux: Root Reducer
const rootReducer = combineReducers({
	form: formReducer
});


// Middleware: Redux Thunk (Async/Await)
const middleware: any = [thunk];

// Middleware: Redux Logger (For Development)
if (process.env.NODE_ENV !== 'production') {
	middleware.push(createLogger());
}

// Middleware: Redux Persist Config
const persistConfig = {
	// Root?
	key: 'root',
	// Storage Method (React Native)
	storage: AsyncStorage,
	// Whitelist (Save Specific Reducers)
	whitelist: [
		'authReducer',
	],
	// Blacklist (Don't Save Specific Reducers)
	blacklist: [],
};

// Middleware: Redux Persist Persisted Reducer
const persistedReducer = persistReducer(persistConfig, rootReducer)

// Redux: Store
const store = createStore(
	persistedReducer,
	applyMiddleware(...middleware),
);

// Middleware: Redux Persist Persister
let persistor = persistStore(store);

// Exports
export {
	store,
	persistor,
};
