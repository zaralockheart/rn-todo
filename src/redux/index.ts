interface Action<PROPS> {
	type: string
	payload: PROPS
}
