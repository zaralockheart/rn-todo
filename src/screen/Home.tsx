import React from "react"
import {SafeAreaView, StyleSheet, TouchableOpacity, View, Text} from "react-native"
import {useNavigation} from "react-navigation-hooks"

export default () => {

	const {navigate} = useNavigation();

	return (
		<React.Fragment>
			<SafeAreaView>
				<View style={style.appbar}>
					<Text>Home</Text>
					<TouchableOpacity style={style.addButton} onPress={() => navigate('AddTodo')}>
						<View/>
					</TouchableOpacity>
				</View>
			</SafeAreaView>
		</React.Fragment>
	)
}

const style = StyleSheet.create({
	appbar: {
		flexDirection: 'row',
		justifyContent: 'space-between'
	},
	addButton: {
		height: 20,
		width: 20,
		backgroundColor: 'red'
	}
})
