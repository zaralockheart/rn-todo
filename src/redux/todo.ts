import {Dispatch} from "redux"

// Action Names
const ADD_TODO = 'ADD_TODO'
const REMOVE_TODO = 'REMOVE_TODO'

// Actions
interface AddTodoProps {
	title: string
	subtitle?: string
	days: Map<number, String>
}

export const addTodo = (data: AddTodoProps) => (dispatch: Dispatch) => {
	dispatch({
		type: ADD_TODO,
		payload: data
	})
}

interface RemoveTodoProps {
	id: string
}

export const removeTodo = (data: RemoveTodoProps) => (dispatch: Dispatch) => {
	dispatch({
		type: REMOVE_TODO,
		payload: data
	})
}

// All Props
type TodoProps = AddTodoProps & RemoveTodoProps

// Initial State
interface TodoObject {
	id: string
	title: string
	subtitle: string
	days: Map<number, String>
}

const initialState = {
	todo: {} as { [key: string]: TodoObject },
	ids: [] as string[]
}

export type TodoState = typeof initialState

export const signInDetailsReducer = (state = initialState, action: Action<TodoProps>) => {
	switch (action.type) {
		case ADD_TODO:
			break;
		default:
	}
}
