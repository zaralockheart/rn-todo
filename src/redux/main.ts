// Imports: Dependencies
import {applyMiddleware, combineReducers, createStore, Store} from 'redux'
import thunk from 'redux-thunk'
import AsyncStorage from '@react-native-community/async-storage'
import {createLogger} from 'redux-logger'
import {persistReducer, persistStore} from 'redux-persist'
import {FormStateMap, reducer as formReducer} from 'redux-form'
import {composeWithDevTools} from 'redux-devtools-extension'

import {todo, todoState} from './todo'

interface Actions<PROPS> {
    type: string
    payload: PROPS
}

interface ReducersAugment {
    form: FormStateMap
    todo: todoState
}

// Redux: Root Reducer
const rootReducer = combineReducers({
    form: formReducer,
    todo
})

// Middleware: Redux Thunk (Async/Await)
const middleware: [any] = [thunk]

// Middleware: Redux Logger (For Development)
if (process.env.NODE_ENV !== 'production') {
    middleware.push(createLogger())
}

// Middleware: Redux Persist Config
const persistConfig = {
    // Root?
    key: 'root',
    // Storage Method (React Native)
    storage: AsyncStorage,
    // Whitelist (Save Specific Reducers)
    whitelist: ['todo'],
    // Blacklist (Don't Save Specific Reducers)
    blacklist: []
}

// Middleware: Redux Persist Persisted Reducer
const persistedReducer = persistReducer(persistConfig, rootReducer)

// Redux: Store
export type State = ReducersAugment

const store: Store<State> = createStore(
    persistedReducer,
    composeWithDevTools(applyMiddleware(...middleware))
)

// Middleware: Redux Persist Persister
let persistor = persistStore(store)

// Exports
export {middleware, store, persistor, Actions, rootReducer}
