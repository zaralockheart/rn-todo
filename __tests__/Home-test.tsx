/**
 * @format
 */

import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {combineReducers, createStore} from 'redux';
import {Provider} from 'react-redux';
import {reducer as formReducer} from 'redux-form';
import {TouchableOpacity} from "react-native"

import Home from "../src/screen/Home"

const rootReducer = combineReducers({
	form: formReducer,
});

const store = createStore(rootReducer);

it('renders correctly', () => {
	const component = renderer.create(
		<Provider store={store}>
			<Home/>
		</Provider>
	);

	expect(component.toJSON()).toMatchSnapshot()

});
