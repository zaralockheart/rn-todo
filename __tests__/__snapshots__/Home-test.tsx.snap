// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text>
      Home
    </Text>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "red",
          "height": 20,
          "opacity": 1,
          "width": 20,
        }
      }
    >
      <View />
    </View>
  </View>
</RCTSafeAreaView>
`;
